clarified expected user input upon config. create empty webdav entries in yaml if use_webdav==false

christian-leroy opened the pull request:
I'm just starting to get familiar with the project. I changed two very minor details regarding the config process:
1. I added two lines clarifying that the user doesn't have to input Zotero/subfolder but just the name of the subfolder. This might be related to #16, but I'm not sure.
2. If WebDAV is not used, the required fields for WebDAV using in the config are added as empty. I switched from Zotero storage to WebDAV and had to look up the fields in `write_config`. Having them as empty in the config might save someone some time. :)

Azeirah commented:
Given that there are probably other people with `Zotero/read` and `Zotero/unread` in their configs, it's probably better to robustly handle that specific case rather than expect people to set the data correctly.

Additionally, I think it's probably better to just use a `null` value or `false` if yaml doesn't have `null` for the webDAV config values. Or perhaps not set them at all!

christian-leroy commented:
1. I agree! I will implement this.
2. How about simply adding a comment at the end of the YAML saying something like: If you want to switch to WebDAV you will have to set these 3 values. I think it should be documented in the config what needs to be changed if the user wants to switch to WebDAV.

Azeirah commented:
> 1. I agree! I will implement this.
> 
>     2. How about simply adding a comment at the end of the YAML saying something like: If you want to switch to WebDAV you will have to set these 3 values. I think it should be documented in the config what needs to be changed if the user wants to switch to WebDAV.

That's perfectly reasonable.

I often see this kind of pattern in various projects which use config files:

```
## Uncomment the following if you want to use ...
# some_variable = 3
```

christian-leroy commented:
Here are my changes:
1. Implemented a simple `normalize_remarkable_path` function alongside some test cases. 
2. Changed the `if` statement in `write_config` to `if config_data["USE_WEBDAV"].lower() != "true"` to ensure case-insensitivity.
3. If user input for `USE_WEBDAV` is anything but `true`, the comments are written at the end of the YAML file as discussed.

I'm aware that #4 is still open. Regardless I added some simple `pytest` asserts to make sure `normalize_remarkable_path` works as intended. If this is not wanted we can simply remove the tests/* folder without any impact.

Azeirah commented:
> Here are my changes:
> 
>     1. Implemented a simple `normalize_remarkable_path` function alongside some test cases.
> 
>     2. Changed the `if` statement in `write_config` to `if config_data["USE_WEBDAV"].lower() != "true"` to ensure case-insensitivity.
> 
>     3. If user input for `USE_WEBDAV` is anything but `true`, the comments are written at the end of the YAML file as discussed.
> 
> 
> I'm aware that #4 is still open. Regardless I added some simple `pytest` asserts to make sure `normalize_remarkable_path` works as intended. If this is not wanted we can simply remove the tests/* folder without any impact.

Nice test!

What I meant with

> Given that there are probably other people with Zotero/read and Zotero/unread in their configs, it's probably better to robustly handle that specific case rather than expect people to set the data correctly.

is that there are also people who have this kind of config _currently_, and we should also handle this when the configuration is already created before syncing. Not just when the configuration is initially set.

So the `normalize_rm_path` should also be used when reading the path I think

Azeirah commented:
Maybe it's overkill to handle it at runtime and we should just give a clear error and point to the right kind of configuration if the path starts with "zotero/", that's also a possibility.

That being said, using the `normalize_rm_path` function is fine anyway, because it exists now.

christian-leroy commented:
Our approach might be a bit unnatural. I think it SHOULD say `Zotero/read` in the config, because that's the actual path. Upon first config it was my instinct to enter `Zotero/read` and others probably feel the same. It's the logical thing to do – and it points the user towards checking if they actually set up their RM correctly, too. 

Also: Is there a particular reason the parent folder has to be named `Zotero`? Maybe I'm overthinking this right now, but why not handle this in the config: Instead of doing things like `if rmapi.upload_file(file_name, f"/Zotero/{folders['unread']}"):` we put the comlete path in our config. I see two benefits: 1. The top folder could be named whatever you want. 2. This potentially opens the door for having multiple folders to be synced.

This would obviously mean getting a little bit deeper into the code. As a start, it's fine to handle the wrong config settings at runtime (or point to the source of the error) – the main goal of my PR is to avoid having sync not work without any useful feedback, as seen in #16.

Azeirah commented:
> Also: Is there a particular reason the parent folder has to be named Zotero?

No. There isn't. This is just how it was since the start. It slipped into being hardcoded as `"/Zotero/" + config_folder`. I agree that it would be better to have the full path being configurable.

> This potentially opens the door for having multiple folders to be synced.

How do you see that working? In principle, the workflow right now is that you always _start_ by adding a sync tag in Zotero, and then it gets synced to rM.

Are you imagining a workflow where things you do in the reMarkable end up in some kind of an "unsorted" or otherwise named collection in Zotero? I suppose that could be useful for "sketches" and such 🤔

Azeirah commented:
> I imagine something like: You're working on two projects, p1 and p2 and you want to have two folders on RM where the literature for each project ends up. So you create p1/read, p1/unread, p2/read, p2/unread on your device. In Zotero you specify with tags such as to_sync_[FOLDERNAME] to which folder this file should be synced. This could be a one-to-many relationship where one file belongs to more than one project – not sure if there's a use case for that, though.

Hmm, maybe. If there's demand for something like this, or what I mentioned then it will at least be easier to implement. We'll see.

> Another suggestion: We could have a default config of `Zotero/unread` and `Zotero/read` in any case. If you don't specify anything in the config – or fail to specify correctly – we could sync to those folders instead with `rmapi` checking if they exist and if not create them. This would be a preferable way to do the initial setup, too, I suppose.

This sounds good to me, to be honest.

Keep in mind we also need to include instructions for the user to create these folders on the reMarkable themselves. I'm not a fan of automatically creating folders on a device as personal as this. So if someone opts to pick different folders, they need to be instructed to create those.
